src/main/java/bio/terra/testrunner/common/utils/KubernetesClientUtils.java

```diff
@@ -582,7 +582,14 @@ public static void changeReplicaSetSizeAndWait(
       int podCount, String componentLabel, String apiComponentLabel) throws Exception {
     V1Deployment apiDeployment = getApiDeployment(componentLabel, apiComponentLabel);
     if (apiDeployment == null) {
-      throw new RuntimeException("API deployment not found.");
+      // If the test configuration does not provide the component label,
+      // this will default to the 'api' value.
+      //
+      // If either the provided component label or default 'api' label
+      // does not exist in the k8s deployment metadata, then issue a
+      // warning and simply exit without doing anything.
+      logger.warn("Application component label '{}' was not found", apiComponentLabel);
+      return;
```

Comment:
Why change this from throwing an exception to just logging? I think this means that `changeReplicaSetSizeAndWait` can just fail to update the replica set size, but the test won't throw an error. That seems confusing because test runner is not doing what the config specifies.

Reply:
The `changeReplicaSetSizeAndWait` gets executed when the `skipKubernetes` flag is set to `false` in the Server Specification. When the integration, perf and resiliency tests were consolidated into one GitHub Action workflow sharing essentially the same `action.yml` and Server Specification, Test Runner will try execute `changeReplicaSetSizeAndWait` with the default 'api' component label (because the integration / perf tests didn't have to specify the application component label in the test config) causing an exception to be thrown and stop the test. So I changed the code to prevent the exception.

Reply:
Do we want the integration/perf tests to touch kubernetes (e.g. set the replica set size before running each test configuration)?

Reply:
I recall a prior discussion that we should still allow integration/perf tests to have access to Kubernetes. By default, if replica set size was not specified in test config, Test Runner will default to 1 (of course only if the application component label is valid).